File: rs/hold/pushtovar/fs/cfgadapt/FooSflCfgAdapter.kt
package pushtovar.fs.cfgadapt

import pushtovar.config.AppCfgInfo
import pushtovar.fs.FooSflCfgInfo
import pushtovar.fs.fooSflCfgSrc
import pushtovar.fwk.CfgSrcAdaptation

fun fooSflCfgAdapter(appCfg: AppCfgInfo): FooSflCfgInfo {
	return FooSflCfgInfo(
        x = appCfg.x,
    )
}

val fooSflCfgAdaptation = CfgSrcAdaptation(
	fooSflCfgSrc,
	::fooSflCfgAdapter,
)

File: rs/pushcfgtovar/src/fs/cfgadapt/foo_sfl_cfg_adapter.rs
use crate::fs::foo_sfl::{FooSflCfgInfo, FOO_SFL_CFG_SRC};
use crate::fwk::{adapt_by_ref, RefreshMode};
use common::config::AppCfgInfo;

pub fn foo_sfl_cfg_adapter(app_cfg: &AppCfgInfo) -> FooSflCfgInfo {
    FooSflCfgInfo {
        a: app_cfg.x.clone(),
        b: app_cfg.y,
    }
}

pub fn foo_sfl_adapt_cfg_src(
    origin: impl Fn() -> AppCfgInfo + 'static + Send + Sync,
    refresh_mode: RefreshMode,
) {
    adapt_by_ref(origin, foo_sfl_cfg_adapter, refresh_mode, &FOO_SFL_CFG_SRC);
}
